data_steward/cdr_cleaner/cleaning_rules/ensure_date_datetime_consistency.py

```diff
@@ -46,62 +62,92 @@
+        """
+        Initialize the class with proper information.
+
+        Set the issue numbers, description and affected datasets. As other tickets may affect this SQL,
+        append them to the list of Jira Issues.
+        DO NOT REMOVE ORIGINAL JIRA ISSUE NUMBERS!
+        """
+        desc = (
+            'Ensure date and datetime consistency where nulls exist in nullable fields'
+            'by setting date in datetime to the date in date field and time as midnight (00,00,00)'
+        )
+        super().__init__(issue_numbers=['DC-614', 'DC-509', 'DC-432'],
+                         description=desc,
+                         affected_datasets=[
+                             cdr_consts.RDR, cdr_consts.UNIONED,
+                             cdr_consts.COMBINED
+                         ],
+                         project_id=project_id,
+                         dataset_id=dataset_id,
+                         sandbox_dataset_id=sandbox_dataset_id)
+
+    def get_cols(self, table):
+        """
+        Generates the fields to choose along with case statements to generate datetime
+        And ensure no null datetime values
+
+        :param self: instance of EnsureDateDatetimeConsistency class
+        :param table: table for which the fields are pulled
+        :return: cols
+        """
+        table_fields = field_mapping.get_domain_fields(table)
+        col_exprs = []
+        for field in table_fields:
+            if field in TABLE_DATES[table]:
+                col_expr = FIX_NULL_OR_INCORRECT_DATETIME_QUERY.format(
+                    field=field, date_field=TABLE_DATES[table][field])
+            else:
+                col_expr = field
+            col_exprs.append(col_expr)
+        cols = ', '.join(col_exprs)
+        return cols
+
+    def get_query_specs(self, project_id, dataset_id):
+        """
+        This function generates a list of query dicts for ensuring the dates and datetimes are consistent
+
+        :param project_id: the project_id in which the query is run
+        :param dataset_id: the dataset_id in which the query is run
+        :return: a list of query dicts for ensuring the dates and datetimes are consistent
+        """
+        queries = []
+        for table in TABLE_DATES:
+            query = dict()
+            query[cdr_consts.QUERY] = FIX_DATETIME_QUERY.format(
+                project_id=self.get_project_id(),
+                dataset_id=self.get_dataset_id(),
+                table_id=table,
+                cols=self.get_cols(table))
+            query[cdr_consts.DESTINATION_TABLE] = table
+            query[cdr_consts.DISPOSITION] = bq_consts.WRITE_TRUNCATE
+            query[cdr_consts.DESTINATION_DATASET] = self.get_dataset_id()
+            queries.append(query)
+        return queries
+
+    def setup_rule(self):
+        """
+        Function to run any data upload options before executing a query.
+        """
+        pass
```

Comment:
@dcampbell-vumc you may want to add

```
def get_sandbox_tablenames(self):
        """
        Returns an empty list because this rule does not use sandbox tables.
        """
       return []
```

to your class.  I added this as an abstract method to the base class.  so when the patch for dc-734 gets added, this will be needed.  you can add it now as part of this patch without any bad side effects.

Reply:
Okay, I added it.

Reply:
Great!  Thank you!